### PR TITLE
Group ranking

### PR DESCRIPTION
**Pull request by vpontis**

### CODE REVIEW

**Comment by vpontis** [comment]
check this out @stubailo

File: app/controllers/groups_controller.rb [hunk 1]
```diff
@@ -40,9 +40,6 @@ def add_player
     end
   end
 
-  def edit
```
Comment (stubailo) [inline app/controllers/groups_controller.rb:43]: this needs to be here still - admins can change group name

Comment (vpontis) [inline app/controllers/groups_controller.rb:43]: The edit method doesn't actually do anything. The page still works.

File: app/models/game.rb [hunk 2]
```diff
@@ -13,6 +13,14 @@ def confirmed?
   def confirm
     self.confirmed=true
     self.transfer_points
+
+    shared_groups = self.winners[0].groups & self.winners[1].groups
```
Comment (stubailo) [inline app/models/game.rb]: This is kinda weirdly written

Comment (vpontis) [inline app/models/game.rb]: yea i'll try to clean it up

File: app/models/group.rb [hunk 3]
```diff
@@ -39,7 +36,7 @@ def add_player_by_username(username)
   end
 
   def ranked_players
-    return players.sort {|a,b| b.group_points(self) <=> b.group_points(self) }
+    return group_players.sort {|a,b| b.points <=> a.points }
```
Comment (stubailo) [inline app/models/group.rb]: isn't this just going to return the join model and not the actual player objects?

File: app/models/group.rb [hunk 4]
```diff
@@ -61,28 +58,25 @@ def game_in_group?(game)
...
-    # recalculate the score of all players in the group
-    def calculate_player_rankings
-      self.group_players.each { |p| p.points = 1000; p.save; }
-
-      self.games.each do |game|
-        group_game_winners = game.winners.map {|winner| self.group_players.find_by(player_id: winner.id)}
-        group_game_losers = game.losers.map {|loser| self.group_players.find_by(player_id: loser.id)}
+  # transfers points for group players in a game
+  def transfer_points(game)
+    group_game_winners = game.winners.map {|winner| self.group_players.find_by(player_id: winner.id)}
```
Comment (stubailo) [inline app/models/group.rb:61]: self.group_players.where(player_id: game.winner_ids)

I think that should work and is better

File: app/models/player.rb [hunk 5]
```diff
@@ -111,4 +111,14 @@ def self.find_first_by_auth_conditions(warden_conditions)
       where(conditions).first
     end
   end
+
+  private
+    def all_players_in_group(game, group)
```
Comment (stubailo) [inline app/models/player.rb]: is this method used anywhere?

File: app/views/groups/show.html.haml [hunk 6]
```diff
@@ -14,12 +14,9 @@
 .row
   .col-sm-6
     %h3 Players
-    - @group.players.each do |player|
-      .player= link_to player.username, player, :class => "player-link"
-    - if can? :add_player, @group
-      = render 'add_player_form'
+    = render partial: 'players/ranking', locals: { players: @group_players }
```
Comment (stubailo) [inline app/views/groups/show.html.haml]: Is the add player form gone now?

Comment (vpontis) [inline app/views/groups/show.html.haml]: ah i didn't mean to delete that

## 

VPontis

File: app/controllers/groups_controller.rb (hunk 1 above)
Comment (stubailo) [inline app/controllers/groups_controller.rb:43]: Weird, I had no idea!

File: app/models/game.rb [hunk 7]
```diff
@@ -13,6 +13,13 @@ def confirmed?
   def confirm
     self.confirmed=true
     self.transfer_points
+
+    shared_groups = self.groups_in_common
```
Comment (stubailo) [inline app/models/game.rb]: no reason to have this extra variable?

File: app/models/group.rb [hunk 8]
```diff
@@ -67,28 +64,29 @@ def game_in_group?(game)
...
+      winner.save
+    end
+    group_game_losers.each do |loser|
+      loser.points -= point_change
+      loser.games_count += 1
+      loser.save
     end
+  end
+
+  def self.global
```
Comment (stubailo) [inline app/models/group.rb]: maybe name this find_global? it's not obvious from the name that it accesses DB

**Pull request merged**